Lib/uuid.py

```diff
@@ -674,15 +675,22 @@ def uuid1(node=None, clock_seq=None):
             is_safe = SafeUUID.unknown
         return UUID(bytes=uuid_time, is_safe=is_safe)
 
-    global _last_timestamp
+    global _last_timestamp, _last_clock_req
     import time
     nanoseconds = time.time_ns()
     # 0x01b21dd213814000 is the number of 100-ns intervals between the
     # UUID epoch 1582-10-15 00:00:00 and the Unix epoch 1970-01-01 00:00:00.
     timestamp = nanoseconds // 100 + 0x01b21dd213814000
     if _last_timestamp is not None and timestamp <= _last_timestamp:
-        timestamp = _last_timestamp + 1
+        # if there is a previous clock sequence, then increment
+        # otherwise, the clock sequence will be regenerated
+        if _last_clock_req is not None:
+            clock_seq += 1
```

Comment:
This wouldn't work. You need to change the last clock sequence not the one that is given as input.

Reply:
This is executed when the clock triggers a rollback, it is correct because detecting the clock rollback needs to happen before the update.

Reply:
I got misunderstood: the problem is that you are incrementing the parameter that is given as input, which can be None. You need to work with the last clock sequence value instead.

Reply:
> I got misunderstood: the problem is that you are incrementing the parameter that is given as input, which can be None. You need to work with the last clock sequence value instead.

Yes, I noticed that I was missing the statement to modify the current clock sequence.